element/index.js

```diff
@@ -65,3 +66,35 @@ export function renderToString( element ) {
 
 	return renderToStaticMarkup( element );
 }
+
+/**
+ * Concat two React children objects
+ *
+ * @param  {?Object} children1 First children value
+ * @param  {?Object} children2 Second children value
+ *
+ * @return {Array}             The concatenation
+ */
+export function concatChildren( children1, children2 ) {
```

Comment:
I was poking around at this function, and had a couple observations:

- I think it'd be easy enough and a nice feature to support a spread value of `...childrens`, so two _or more_ set of children can be concatenated
- I'm still unclear why `toArray` need exist. You'd mentioned different values, but [in my testing](http://jsbin.com/siwezosixa/edit?html,js,console) `Children.toArray` seems to handle most everything thrown at it.
- I think we might be able to drop an iteration by creating the return value _while_ iterating children (instead of `toArray` then `map`)

Do you think something like this would work?

```js
/**
 * Concatenate two or more React children objects
 *
 * @param  {...?Object} childrens Set of children to concatenate
 * @return {Array}                The concatenated value
 */
export function concatChildren( ...childrens ) {
	return childrens.reduce( ( memo, children, i ) => {
		Children.forEach( children, ( child, l ) => {
			memo.push( cloneElement( child, {
				key: [ i, l ].join()
			} ) );
		} );

		return memo;
	}, [] );
}
```